src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -148,8 +154,15 @@ class ClusterMonitorActor(val cluster: Cluster,
     * @return ShutdownActor
     */
   private def handleFailedCluster(errorDetails: ClusterErrorDetails): Future[ClusterMonitorMessage] = {
-    // Delete the cluster in Google
-    gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName).flatMap { _ =>
+    val deleteFuture = for {
+      // Delete the cluster in Google
+      _ <- gdDAO.deleteCluster(cluster.googleProject, cluster.clusterName)
+      // Remove the Dataproc Worker IAM role for the pet service account
+      _ <- googleIamDAO.removeIamRolesForUser(cluster.googleProject.string, cluster.googleServiceAccount, Set("roles/dataproc.worker"))
```

Comment:
Do we want this to happen even if `deleteCluster()` fails?

Reply:
Yeah probably.. will fix